### ISSUE TITLE
Headless VSCode means keyring dialogs wont pop up

### ISSUE THREAD

**Issue by bporterfield** [looker-open-source/malloy#258]
From Jonathan:
it was just failing to set to the keyring because it didn't have access
or was requesting access from the OS
but of course because I was running headless
it wasn't showing me that "unlock keyring" dialog
but when I opened CRD it was working

Not sure if there's much we can do, but perhaps we can warn the user?

**Comment by jonathanswenson** [comment]
I was using vscode in the [ssh / headless mode](https://code.visualstudio.com/docs/remote/ssh) and every time I tried to click save on a postgres connection that required a password it just hid the “save” button but didn’t actually do anything -- no warning, no error. 

I went to try it using a normal “headful” vscode in that environment, and noticed that I had permission request sitting open on the desktop when I remoted in. 

![image](https://user-images.githubusercontent.com/2752979/150930924-c44a8125-7475-47ea-b70a-86762bc8dc69.png)

After approving access to the keyring , it worked for me without any issues.

**Comment by lloydtabb** [comment]
Not sure how we fix this one.  The Keyring UI is controlled by the OS.  @jonathanswenson you are running code-server?

**Comment by jonathanswenson** [comment]
Yeah I believe it is code-server. I set this up a long time ago, but following the instructions [here](https://code.visualstudio.com/docs/remote/ssh). 

If nothing else, perhaps an error message / timeout message about failure to write to the keyring after it doesn’t succeed for some amount of time? Then someone could at least try to figure out how to fix the keyring?
